lib/checkstl.cpp

```diff
@@ -1245,11 +1246,19 @@ static std::pair<const Token *, const Token *> isMapFind(const Token *tok)
         return {};
     if (!astIsContainer(tok->astOperand1()->astOperand1()))
         return {};
+    const Token * contTok = tok->astOperand1()->astOperand1();
+    std::string typeName = Token::typeStr(contTok);
+    if (typeName.empty())
+        return {};
+    if (typeName.find("multi") != std::string::npos)
+        return {};
+    if (typeName[0] == 'Q')
```

Comment:
This does not feel proper to me. What about other containers. Boost, WxWidgets, MFC, VCL, ...imho you should ensure that it's a `std` container if it rely on very specific semantics that the `std` containers have.

Reply:
> What about other containers. Boost, WxWidgets, MFC, VCL, 

It works with boost and wxwidgets, and MFC is not a problem because the names dont overlap. I dont know what VLC is.

> ou should ensure that it's a std container if it rely on very specific semantics that the std containers have.

It doesnt just rely semantics of `std` containers. It relies on C++'s requirements for class to considered an `AssociativeContainer`. Most C++ libraries follow these type requirements like boost, wxwidgets and even more such as thread building blocks.

We should add an attribute to the library to know whether it follows C++'s requirements for containers or not, but I am not sure how to do that.

Reply:
> I dont know what VLC is.

VCL is the `Visual Component Library` used in C++ Builder.

> It works with boost and wxwidgets, and MFC is not a problem because the names dont overlap.

ok so you're saying Qt is a rare exception. Then I guess we can have this code.

Reply:
> I dont know what VLC is.

For information VCL is `Visual Component Library` used in C++ Builder.

> It relies on C++'s requirements for class to considered an AssociativeContainer.

ok.

> It would be better to have an attribute to the library to know whether it follows C++'s requirements for containers or not, but I am not sure how to do that.

It is not hard. You update the xml parsing in `library.cpp`, this is the hardest part. Then use the attribute in the relevant cfg files. The `cfg/cppcheck-cfg.rng` should be updated. Some documentation should be added in `man/manual.md`.

Reply:
> ok so you're saying Qt is a rare exception.

Yes.